node/src/validator/mod.rs

```diff
@@ -115,21 +127,48 @@ impl<N: Network, C: ConsensusStorage<N>> Validator<N, C> {
         // Initialize the signal handler.
         node.handle_signals();
 
-        // Start the BFT consensus here
-        let bft = BftConsensus::new(consensus, router, dev)?;
-        let (primary, worker) = bft.start().await.unwrap();
+        // TODO: isolate the BFT logic below to a new method that is started only once the validator mesh is ready.
+
+        // Prepare the path containing BFT consensus files.
+        let bft_path =
+            format!("{}/node/bft-consensus/committee/{}", workspace_dir(), if dev.is_some() { ".dev" } else { "" });
 
-        // Start the primary.
-        let handle_primary = tokio::spawn(async move {
-            primary.wait().await;
```

Comment:
where did these `wait()` calls go?

Reply:
I noticed that once you call `start` ([L169](https://github.com/eqlabs/snarkOS/pull/14/files/9e8fc28a76878a0cfa0f1eca75ecbb3cce6837ca#diff-815512666511928f910f9f253a7c2385374af110a5f308dc15a5c563319607ddR169)) the consensus is already running; and if we notice that we need them, they should go into the aforementioned method to trigger always, both in full validators and consensus-only tests.

Reply:
I think it's used in the test app to wait for completion?

Reply:
Then I'll put it on my TODO to see where it would fit best, but preferably it would be self-contained within the consensus objects :+1:.